pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -164,7 +165,7 @@ def _copy_trainer_model_properties(self, model):
 
         for m in [model, ref_model]:
             m.trainer = proxy(self.trainer)
-            m.use_amp = self.trainer.amp_backend is not None
```

Comment:
Deepspeed could use amp, but it's not extended from MixedPrecisionPlugin
@daniellepintz's change should fix this https://github.com/PyTorchLightning/pytorch-lightning/pull/12323